src/nvim/ops.c

```diff
@@ -4643,15 +4654,11 @@ void write_reg_contents_lst(int name, char_u **strings, int maxlen,
 /// @param block_len width of visual block
 void write_reg_contents_ex(int name,
                            const char_u *str,
-                           ssize_t len,
+                           size_t len,
                            bool must_append,
                            int yank_type,
                            long block_len)
 {
-  if (len < 0) {
```

Comment:
it seems you have handled all cases this affects, but I'm not sure we really want this. `-1` for an unlimited length seems to be a common idiom in the codebase, and I don't know if we want to get rid of it. Anyway it seems unrelated to enabling `-Wconversion` ?